### ISSUE TITLE
Video pans up and to the left

### ISSUE THREAD

**Issue by bourneash** [Animator-Anon/animator_extension#29]
Hey! It seems like no matter what settings I put in, my videos seem to pan up and to the left every time. I have tried setting a specific seed, using a initial image, and even setting the keyframe like this `0 | perspective | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0`.

I'm sure I am missing something, any help would be appreciated. Thanks!

**Comment by Animator-Anon** [comment]
What are your keyframes, the transform ones? Were you trying to use perspective? It was never fully tested, just something I mucked around with one day.

**Comment by bourneash** [comment]
Maybe I am not understanding the keyframe concept. For example, I'd like to generate an image that morphs from face to face. For example, I've tried this keyframe prompt for a 10 second video. 

0 | seed | 3988832402
0 | prompt | shaman, vibrant colors, highly detailed, forrest, cinematic, <lora:xal:0.6> | easynegative
10   | transform | 0   |    0 |    0 |   0

In this example I have the seed set to -1 and no prompts in the GUI, just in the keyframes.

**Comment by Animator-Anon** [comment]
Did you have a look at the examples in the readme for the repository? They might make things clearer.

**Comment by bourneash** [comment]
Yes, but I am still rather confused. Do the keyframe's effect the time before or after it is hit by the time_s

So, for example:
`0 | seed | 3988832402
0 | prompt | male shaman, vibrant colors, highly detailed, forrest, cinematic, lora:xal:0.6 | easynegative
10 | prompt | female shaman, vibrant colors, highly detailed, forrest, cinematic, lora:xal:0.6 | easynegative`

Let's say I'd like to make a video that transforms a male shaman to a female shaman within a 10 second duration.
Would the "female" `prompt` in this example be applied to the content during the time_s from 0s - 10s, or will it only be applied to the content after the 10 second mark (and never morph to  female since the video is 10s long)?

**Comment by Animator-Anon** [comment]
This can get pretty complex, lets focus on prompts for now. 

There are two main operation modes for animator. Loopback mode, typically what the script used to do, generates an image and feeds it back into SD. It is like img2img over and over, and what is typically used to do animations.
The other mode just generates an image based on the current prompts and seed, like txt2img.

At each frame, the prompt is made up of three different things. A style if you have that set, the template, and finally the most recent set of prompts.

In your example, in loopback mode, if you have prompt interpolation turned off, at frame 0 you will get that prompt. You will keep getting that prompt until time 10s. Then the new prompt will take over and the image will slowly change to the new prompt.
If you have prompt interpolation turned on, at time 0, you will get 100% of the first prompt and 0% of the time 10s prompt. The numbers will change until time 10, where you get 0% first prompt and 100% of the second prompt. But you are still in loopback mode, so this is only influencing the initial picture.
If you turn loopback mode off, leave prompt interpolation on, it will generate clean images every time with a mix of both prompts, from the first set to the second. But the results are not as you might expect, this comes down to how the SD system works I suppose.

The best way to get what you want is probably do something like this:
Loopback mode: on
prompt interpolation: off

0 | template | vibrant colors, highly detailed, forrest, cinematic | some negative prompts
0 | prompt | male shaman |
1 | prompt | female shaman |

This will mean for the first second, you get pure male shaman, and then from seconds 2-10, it will start changing the image to female shaman. How fast it does this depends on the denoise strength.
